### PR TITLE
feature: Support CICS translator options

### PR DESCRIPTION
**Pull request by ap891843**
Support CICS translator options. For more details visit  https://www.ibm.com/docs/en/cics-ts/6.1?topic=compilation-using-cics-translator

Signed-off-by: ap891843 <aman.prashant@broadcom.com>

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectService.java [hunk 1]
```diff
@@ -50,6 +51,9 @@ public DialectService(CopybookService copybookService,
 
     dialect = new DaCoDialect(messageService, new DaCoMaidProcessor(copybookService, treeListener, messageService));
     dialectSuppliers.put(dialect.getName(), dialect);
+
+    CICSTranslatorDialect cicsTranslatorDialect = new CICSTranslatorDialect(messageService);
```
Comment (Nurkambay) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectService.java:55]:
```
 dialect = new CICSTranslatorDialect(messageService);
```

File: server/engine/src/main/java/org/eclipse/lsp/cobol/service/CachingConfigurationService.java [hunk 2]
```diff
@@ -110,7 +110,10 @@ private SQLBackend parseSQLBackend(List<Object> objects) {
   }
 
   private List<String> parseDialects(JsonArray dialects) {
-    return Streams.stream(dialects).map(JsonElement::getAsString).collect(toList());
+    List<String> providedDialects = Streams.stream(dialects).map(JsonElement::getAsString).collect(toList());
+    if (!providedDialects.contains(CICS_TRANSLATOR_DIALECT))
+      providedDialects.add(CICS_TRANSLATOR_DIALECT);
```
Comment (Nurkambay) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/service/CachingConfigurationService.java:115]: Shouldn't this dialect be the 1st?

File: server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/CompletionStorage.java [hunk 3]
```diff
@@ -95,9 +92,10 @@ private void fillInStorage(Map<String, T> props) {
 
   private void setDialects(List<Object> dialectObject) {
     JsonArray jsonObj = (JsonArray) dialectObject.get(0);
-    this.dialectList = Streams.stream(jsonObj).map(JsonElement::getAsString).collect(toList());
+    List<String> dialectList = Streams.stream(jsonObj).map(JsonElement::getAsString).collect(toList());
     if (!dialectList.isEmpty())
       this.dialectType = dialectList.contains(DaCoDialect.NAME) ? DaCoDialect.NAME : IdmsDialect.NAME;
+    dialectList.add(CICS_TRANSLATOR_DIALECT);
```
Comment (Nurkambay) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/CompletionStorage.java:98]: Looks like this list is not used anywhere.

**Review by Nurkambay — CHANGES_REQUESTED** [review]

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectService.java (hunk 1 above)
Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectService.java]: updated

File: server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/CompletionStorage.java (hunk 3 above)
Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/CompletionStorage.java]: Updated. It is not needed

**Review by Nurkambay — APPROVED** [review]

**Pull request merged**